### PR TITLE
fix: LoginPage for wide screens

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

Use KeyboardAvoidingView to push the input into the viewport and Scrollview to Allow the user to scroll the screen.

### Fixed Issues

$ https://github.com/Expensify/App/issues/5159

### Tests 
1. Open the Login page on a tablet or Ipad.
2. Click on login input and see it remains in the View and is not hidden by Keyboard.
---
1. Open the Password Page on Tablet or Ipad.
2. Clicked all the inputs to see if they are not hidden behind the keyboard.
---
1. Open the Login page on landscape Mode.
2. Clicked all the inputs to see if they are not hidden behind the keyboard.

### QA Steps
1. Open Login, Password pages on Landscape Mode or tablet or Ipad.
2. Observe the input should not be hidden behind the keyboard.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web | Desktop

image

#### Mobile Web

Screenshot 2021-12-07 01:42:23

#### Android

Screenshot 2021-12-07 01:37:36

https://user-images.githubusercontent.com/24370807/144915095-08fd1ae3-215c-4130-ad09-66cb71d808cc.mp4

### CODE REVIEW

File: src/pages/signin/SignInPageLayout/SignInPageContent.js [hunk 1]
```diff
@@ -26,31 +27,35 @@ const propTypes = {
...
-        contentContainerStyle={scrollViewContentContainerStyles}
     >
-        <Form style={[styles.flex1, styles.signInPageNarrowContentContainer, styles.alignSelfStretch, styles.ph5]}>
+        <Form style={[
+            styles.flex1,
+            styles.alignSelfStretch,
+            props.isSmallScreenWidth && styles.signInPageNarrowContentContainer,
+            props.isSmallScreenWidth ? styles.ph5 : styles.ph4,
+        ]}
+        >
```
Comment (Luke9389) [inline src/pages/signin/SignInPageLayout/SignInPageContent.js:54]: Indentation looks a little weird here. Maybe `]}>`?

File: src/pages/signin/SignInPageLayout/index.js [hunk 2]
```diff
@@ -18,26 +18,22 @@ const propTypes = {
...
+        >
+            {props.children}
+        </SignInPageContent>
+    );
+
+    if (props.isSmallScreenWidth) {
+        return content;
+    }
+
+    return <SignInPageWideContainer>{content}</SignInPageWideContainer>;
```
Comment (Luke9389) [inline src/pages/signin/SignInPageLayout/index.js:35]: Is there really an advantage to having the `SignInPageWideContainer` be its own component? I think it'd be better for us to get rid of the `SignInPageWideContainer` altogether and have the View and SVG components it contains live here in this file instead. In other words, why have these as two files when it's not really complicated to have them be one?

**Review by Luke9389 — CHANGES_REQUESTED** [review]

File: src/pages/signin/SignInPageLayout/SignInPageContent.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/signin/SignInPageLayout/SignInPageContent.js:54]: Actually, it's totally fine. There are two ways here. 
This one 
 e.g. https://github.com/Expensify/App/blob/5362f017468a300b263d35bb257102e7d45c9ca8/src/pages/home/report/ReportActionCompose.js#L499-L506
and
```
        <Form
            style={[
                styles.flex1,
                styles.alignSelfStretch,
                props.isSmallScreenWidth && styles.signInPageNarrowContentContainer,
                props.isSmallScreenWidth ? styles.ph5 : styles.ph4,
            ]}
        >
```

File: src/pages/signin/SignInPageLayout/index.js (hunk 2 above)
Comment (parasharrajat) [reply src/pages/signin/SignInPageLayout/index.js:35]: It looks clean this way but let me know I can merge.

Comment (parasharrajat) [reply src/pages/signin/SignInPageLayout/index.js]: I just merged. Thanks for the suggestion.

File: src/pages/signin/SignInPageLayout/SignInPageContent.js (hunk 1 above)
Comment (Luke9389) [reply src/pages/signin/SignInPageLayout/SignInPageContent.js:54]: Huh, Ok well that's fine with me. This was a NAB.

**Review by Luke9389 — APPROVED** [review]

**Pull request merged**